lazy_theta_star_p_planner/src/lazy_theta_star_p.cpp

```diff
@@ -0,0 +1,262 @@
+//  Copyright 2020 Anshumaan Singh
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http:// www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lazy_theta_star_p_planner/lazy_theta_star_p.hpp"
+#include <vector>
+
+namespace lazyThetaStarP
+{
+
+LazyThetaStarP::LazyThetaStarP()
+{
+  costmap_tolerance_ = 1.75;
+  euc_tolerance_2_ = 1.0;
+  euc_tolerance_ = 1.0;
+  size_x = 0;
+  size_y = 0;
+  index_generated = 0;
+}
+
+bool LazyThetaStarP::generatePath(std::vector<coordsW> & raw_path)
+{
+  setContainers();
+  RCLCPP_INFO(node_->get_logger(), "Path Planning Begins... ");
+
+  index_generated = 0;
+  id curr_id = index_generated;
+  addToNodesData(index_generated);
+  nodes_data[curr_id] = {src.x, src.y, getCost(src.x, src.y), dist(src.x, src.y, dst.x, dst.y),
```

Comment:
Why so much redundancy here? you get cost and distances twice

Reply:
Since it has to be only once for each execution, that is why I didn't create a local variable for it